src/main/java/com/minecolonies/colony/buildings/BuildingFarmer.java

```diff
@@ -1,134 +1,538 @@
+     *
+     * @return a field object.
+     */
+    @Nullable
+    public Field getCurrentField()
+    {
+        return currentField;
+    }
+
+    /**
+     * Sets the field the farmer is currently working on.
+     *
+     * @param currentField the field to work on.
+     */
+    public void setCurrentField(final Field currentField)
+    {
+        this.currentField = currentField;
+    }
+
+    /**
+     * Retrieves a random field to work on for the farmer.
+     *
+     * @return a field to work on.
+     */
+    @Nullable
+    public Field getFieldToWorkOn()
+    {
+        Collections.shuffle(farmerFields);
+        for (@NotNull final Field field : farmerFields)
+        {
+            if (field.needsWork())
+            {
+                currentField = field;
+                return field;
+            }
+        }
+        return null;
+    }
+
+    @NotNull
     @Override
-    public String getSchematicName(){ return FARMER; }
+    public String getSchematicName()
+    {
+        return FARMER;
+    }
 
     @Override
-    public int getMaxBuildingLevel(){ return 3; }
+    public int getMaxBuildingLevel()
+    {
+        return MAX_BUILDING_LEVEL;
+    }
 
+    @NotNull
     @Override
-    public String getJobName(){ return FARMER; }
+    public String getJobName()
+    {
+        return FARMER;
+    }
 
+    @NotNull
     @Override
-    public AbstractJob createJob(CitizenData citizen)
+    public AbstractJob createJob(@NotNull CitizenData citizen)
     {
-        return new JobFarmer(citizen); //TODO Implement Later
+        if (!farmerFields.isEmpty())
+        {
+            for (@NotNull final Field field : farmerFields)
+            {
+                final Field colonyField = getColony().getField(field.getID());
+                if (colonyField != null)
+                {
+                    colonyField.setOwner(citizen.getName());
+                }
+                field.setOwner(citizen.getName());
+            }
+        }
+        return new JobFarmer(citizen);
     }
 
     @Override
-    public void readFromNBT(NBTTagCompound compound)
+    public void onDestroyed()
     {
-        super.readFromNBT(compound);
+        super.onDestroyed();
+        for (@NotNull final Field field : farmerFields)
+        {
+            final Field tempField = getColony().getField(field.getID());
 
-        NBTTagCompound farmerCompound = compound.getCompoundTag(TAG_FARMER);
+            if (tempField != null)
+            {
+                tempField.setTaken(false);
+                tempField.setOwner("");
+                @NotNull final ScarecrowTileEntity scarecrowTileEntity = (ScarecrowTileEntity) getColony().getWorld().getTileEntity(field.getID());
+                getColony().getWorld().markBlockForUpdate(scarecrowTileEntity.getPos());
+                scarecrowTileEntity.setName(LanguageHandler.format("com.minecolonies.gui.scarecrow.user", LanguageHandler.format("com.minecolonies.gui.scarecrow.user.noone")));
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Define a constant instead of duplicating this literal "com.minecolonies.gui.scarecrow.user" 3 times. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://home.kk-sc.de:9000/coding_rules#rule_key=squid%3AS1192)